### PR TITLE
[FIX] PLFQ check score types again after reloading IDs.

### PR DESCRIPTION
**Pull request by jpfeuffer**
And add trypticity to search engine parameters.

### CODE REVIEW

File: src/openms/include/OpenMS/METADATA/ProteinIdentification.h [hunk 1]
```diff
@@ -251,6 +251,16 @@ namespace OpenMS
       AVERAGE,
       SIZE_OF_PEAKMASSTYPE
     };
+
+    /// Trypticity
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/METADATA/ProteinIdentification.h:255]: Would it make sense to have this neun only once?

Comment (timosachsenberg) [reply src/openms/include/OpenMS/METADATA/ProteinIdentification.h:255]: neun=enum

**Comment by timosachsenberg** [comment]
hmm or would it better to use something generic instead of NumberTrypticTermini
NumberEnzymaticTermini?

File: src/openms/include/OpenMS/METADATA/ProteinIdentification.h (hunk 1 above)
Comment (jpfeuffer) [reply src/openms/include/OpenMS/METADATA/ProteinIdentification.h:255]: IMHO this should go into the enzyme structures, but this is too much effort right now to support it there.

File: src/openms/include/OpenMS/CHEMISTRY/DigestionEnzyme.h [hunk 2]
```diff
@@ -53,6 +53,19 @@ namespace OpenMS
   {
   public:
 
+    /// Terminus Specificity
+    ///@todo Currently this is just here for encapsulation
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/CHEMISTRY/DigestionEnzyme.h:57]: which I think is totally fine ;)

Comment (timosachsenberg) [reply src/openms/include/OpenMS/CHEMISTRY/DigestionEnzyme.h:57]: but can't we just use the one from EnzymaticDigestion?

File: src/openms/source/ANALYSIS/ID/PeptideIndexing.cpp [hunk 3]
```diff
@@ -51,20 +51,22 @@ using namespace std;
...
 
-    defaults_.setValue("enzyme:specificity", EnzymaticDigestion::NamesOfSpecificity[0], "Specificity of the enzyme."
-                                                                                               "\n  '" + EnzymaticDigestion::NamesOfSpecificity[0] + "': both internal cleavage sites must match."
-                                                                                                                                                     "\n  '" + EnzymaticDigestion::NamesOfSpecificity[1] + "': one of two internal cleavage sites must match."
-                                                                                                                                                                                                           "\n  '" + EnzymaticDigestion::NamesOfSpecificity[2] + "': allow all peptide hits no matter their context. Therefore, the enzyme chosen does not play a role here");
+    defaults_.setValue("enzyme:specificity", "", "Specificity of the enzyme. Default: deduce from input."
+      "\n  '" + EnzymaticDigestion::NamesOfSpecificity[2] + "': both internal cleavage sites must match."
```
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/ID/PeptideIndexing.cpp:62]: hmm on the left we have [0] und here we have[2] could we use the enum as index directly?

File: src/openms/source/CHEMISTRY/EnzymaticDigestion.cpp [hunk 4]
```diff
@@ -80,11 +80,11 @@ namespace OpenMS
 
   EnzymaticDigestion::Specificity EnzymaticDigestion::getSpecificityByName(const String& name)
   {
-    for (Size i = 0; i < SIZE_OF_SPECIFICITY; ++i)
+    for (Size i = 0; i < SPEC_UNKNOWN; ++i)
```
Comment (timosachsenberg) [inline src/openms/source/CHEMISTRY/EnzymaticDigestion.cpp:83]: SIZE_OF_SPECIFICITY is a safer pattern (until we switch to enum classes)

File: src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp [hunk 5]
```diff
@@ -2997,11 +2999,24 @@ namespace OpenMS
...
+        {
+          minCharge = upit->second.toString().toInt();
+        }
+        else if (upit->first == "MaxCharge")
+        {
+          maxCharge = upit->second.toString().toInt();
+        }
+        else if (upit->first == "NumTolerableTermini")
+        {
+          sp.nr_termini = static_cast<EnzymaticDigestion::Specificity>(upit->second.toString().toInt());
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp:3012]: nice fixes

File: src/openms/source/METADATA/ProteinIdentification.cpp [hunk 6]
```diff
@@ -141,7 +142,8 @@ namespace OpenMS
            fragment_mass_tolerance_ppm == rhs.fragment_mass_tolerance_ppm &&
```
Comment (timosachsenberg) [inline src/openms/source/METADATA/ProteinIdentification.cpp:142]: I prefer std::tie constructs if possible

File: src/openms/source/METADATA/ProteinIdentification.cpp [hunk 7]
```diff
@@ -164,7 +166,8 @@ namespace OpenMS
         this->fragment_mass_tolerance_ppm != sp.fragment_mass_tolerance_ppm ||
```
Comment (timosachsenberg) [inline src/openms/source/METADATA/ProteinIdentification.cpp:166]: same here

File: src/openms/include/OpenMS/CHEMISTRY/DigestionEnzyme.h (hunk 2 above)
Comment (jpfeuffer) [reply src/openms/include/OpenMS/CHEMISTRY/DigestionEnzyme.h:57]: yes, forgot to remove

File: src/openms/source/CHEMISTRY/EnzymaticDigestion.cpp (hunk 4 above)
Comment (jpfeuffer) [reply src/openms/source/CHEMISTRY/EnzymaticDigestion.cpp:83]: Add it or use SIZE as unknown?

Comment (timosachsenberg) [reply src/openms/source/CHEMISTRY/EnzymaticDigestion.cpp:83]: I would add SIZE as last and have UNKNOWN extra. At least that's how others do it

File: src/tests/topp/QualityControl_1_out.mzTab [hunk 8]
```diff
@@ -15,7 +15,7 @@ MTD	software[2]-setting[4]	fragment_mass_tolerance_unit:Da
 MTD	software[2]-setting[5]	precursor_mass_tolerance:10.0
 MTD	software[2]-setting[6]	precursor_mass_tolerance_unit:ppm
 MTD	software[2]-setting[7]	enzyme:Trypsin
-MTD	software[2]-setting[8]	nr_enzymatic_termini:10
+MTD	software[2]-setting[8]	nr_enzymatic_termini:3
```
Comment (timosachsenberg) [inline src/tests/topp/QualityControl_1_out.mzTab:18]: 3?

Comment (jpfeuffer) [reply src/tests/topp/QualityControl_1_out.mzTab]: yes, I renamed everything again now anyway.

**Pull request merged**